rotkehlchen/chain/ethereum/modules/convex/balances.py

```diff
@@ -23,24 +19,43 @@
 from .constants import CPT_CONVEX
 
 if TYPE_CHECKING:
+    from rotkehlchen.accounting.structures.evm_event import EvmEvent
+    from rotkehlchen.chain.ethereum.interfaces.balances import BalancesType
     from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import SupportedBlockchain
 
 logger = logging.getLogger(__name__)
 log = RotkehlchenLogsAdapter(logger)
 
-BalancesType = dict[ChecksumEvmAddress, dict[EvmToken, Balance]]
-
 
 class ConvexBalances(ProtocolWithBalance):
     """Query staked balances in the Convex gauges"""
 
-    def __init__(self, database: DBHandler, evm_inquirer: 'EvmNodeInquirer'):
-        super().__init__(database=database, evm_inquirer=evm_inquirer, counterparty=CPT_CONVEX)
+    def __init__(
+            self,
+            database: DBHandler,
+            evm_inquirer: 'EvmNodeInquirer',
+            chain: 'SupportedBlockchain',
+    ):
+        super().__init__(
+            database=database,
+            evm_inquirer=evm_inquirer,
+            chain=chain,
+            counterparty=CPT_CONVEX,
+        )
         self.cvx = A_CVX.resolve_to_evm_token()
 
+    def get_gauge_deposit_events(self) -> set[tuple[HistoryEventType, HistoryEventSubType]]:
+        return {(HistoryEventType.DEPOSIT, HistoryEventSubType.NONE)}
+
+    def get_gauge_address(self, event: 'EvmEvent') -> Optional[ChecksumEvmAddress]:
+        if event.address is None or event.extra_data is None:
+            return None
```

Comment:
why the condition on event.address?

Reply:
For keeping the logic as it was before the abstraction.

Reply:
as I wrote in discord, remove if not needed, or if needed explain why in a comment